src/main/kotlin/dev/rosewood/roseskyblock/manager/IslandManager.kt

```diff
@@ -50,22 +54,17 @@ class IslandManager(rosePlugin: RosePlugin) : Manager(rosePlugin) {
      * @param owner The owner of the islands to load
      * @return the islands that were loaded or the islands from cache
      */
+    @Suppress("UNCHECKED_CAST")
     fun tryLoadIslands(owner: OfflinePlayer): CompletableFuture<Map<IslandWorldGroup, IslandGroup>> {
-        if (this.islandGroups.containsKey(owner.uniqueId))
-            return CompletableFuture.completedFuture(this.islandGroups[owner.uniqueId])
+        if (islandGroups.containsKey(owner.uniqueId)) return CompletableFuture.completedFuture(islandGroups[owner.uniqueId])
 
-        val future = CompletableFuture<Map<IslandWorldGroup, IslandGroup>>()
-        this.rosePlugin.runAsync {
-            val dataManager = this.rosePlugin.getManager(DataManager::class)
-            val islands = mutableMapOf<IslandWorldGroup, IslandGroup>()
-            this.rosePlugin.getManager(WorldManager::class).worldGroups.forEach { worldGroup ->
-                dataManager.getIslandGroup(owner, worldGroup).ifPresent {
-                    islands[worldGroup] = it
-                }
-            }
-            future.complete(islands)
+        return CompletableFuture.supplyAsync {
+            val dataManager = rosePlugin.getManager<DataManager>()
```

Comment:
Hoist dataManager instantiation above the CompletableFuture, it wasn't like this before but I think it would look better

Reply:
It shouldn't be outside of the `CompletableFuture` there